nucypher/blockchain/eth/interfaces.py

```diff
@@ -230,14 +230,17 @@ def get_gas_strategy(cls, gas_strategy: Union[str, Callable]) -> Callable:
         return gas_strategy
 
     def attach_middleware(self):
+        if self.poa is None:  # If POA is not set explicitly, try to autodetect from chain id
+            chain_id = int(self.client.chain_id)
+            self.poa = chain_id in POA_CHAINS
```

Comment:
:+1:

Reply:
As long as the value of `poa is None` from CLI input or configuration files, this auto-detection will work. :+1: